content/manual/os/dojo-tools.md

```diff
@@ -415,3 +497,2958 @@ None.
+  ::
+:-  %say
+|=  [^ [arg=(list path)] vane=?(%g %c)]
+=-  tang+(flop `tang`(zing -))
+.......(truncated for brevity)..........
+```
+
+List the files in a directory:
+
+```
+> +cat %/sys/vane
+  * /~zod/base/~2022.9.4..11.02.01..5b0a/sys/vane
+  ames/hoon
+  behn/hoon
+  clay/hoon
+  dill/hoon
+  eyre/hoon
+  gall/hoon
+  iris/hoon
+  jael/hoon
+  khan/hoon
+```
+
+---
+### `|cp`
+
+Copy a file.
+
+#### Arguments
+
+The first [path][path] is to the file and the second is where to copy it.
+
+```
+input=path output=path
+```
+
+Both the input and output paths must include the full [path prefix][path prefix].
+
+The output path must end with an explicit filename and [mark][mark], you can't
+just point it at a directory.
+
+#### Example
+
+Copy the `code.hoon` generator to the root of `%base` with the name `foo.hoon`:
+
+```
+> |cp %/gen/code/hoon %/foo/hoon
+>=
++ /~zod/base/9/foo/hoon
+```
+
+Let's list the files in the root of `%base`:
+
+```
+> +ls %
+  app/
+  desk/bill
+  foo/hoon   <- new file with contents of %/gen/code/hoon
+  gen/
+  lib/
+  mar/
+  sur/
+  sys/
+  ted/
+```
+
+---
+
+### `=dir`
+
+Change working directory.
+
+Note this is not a [generator][generator] or [thread][thread] as are most other
+tools documented. Rather, it is a special command built directly into the dojo.
+
+#### Arguments
+
+```
+path
+```
+
+The path must include the [path prefix][path prefix].
+
+If no path is specified, it will switch back to the default location (the root
+of the `%base` desk at its most recent revision).
+
+#### Example
+
+Let's try changing to the root of the `%garden` [desk][desk]:
+
+```
+> =dir /=garden=
+=% /~zod/garden/~2022.9.4..13.15.48..5e81
+```
+
+The dojo prompt will now look like:
+
+```
+~zod:dojo/=/garden/~2022.9.4..13.15.48..5e81>
+```
+
+`%` and `/===` now resolve to this location, so any commands like 
+[`+ls %`](#ls), [`+tree %`](#tree), etc, will be run against this location.
+
+The working directory doesn't need to be the root of a desk, you can also do
+`=dir /=base=/gen/hood`, etc.
+
+If you specify a different [case][case] in the path prefix like `=dir /=base/1`,
+you will switch to that revision.
+
+To switch back to the default location (the root of the `%base` desk at its most
+recent revision), you just run `=dir` without an argument.
+
+---
+
+### `+ls`
+
+List files and directories at the specified location.
+
+#### Arguments
+
+The `path` is mandatory, the `vane` is optional:
+
+```
+path, =vane ?(%c %g)
+```
+
+The path must include the [path prefix][path prefix].
+
+The default behavior of `+ls` is to query Clay, but [Gall][gall] [agents][agent]
+can also be queried by specifying the optional `=vane %g` argument like `+ls
+/=some-agent=/foo/bar, =vane %g`. This will perform a [scry][scry] of the form
+`.^(arch %gy /=some-agent=/foo/bar)`. Very few Gall agents implement `%y` scry
+endpoints that produce `arch` types, so this feature is almost entirely useless.
+
+#### Example
+
+```
+> +ls %/sys/vane
```

Comment:
serious question:  should we de-emphasize `%` in favor of `/===`?

Reply:
I personally prefer `%`, I think `/===` is redundant. It's better to think of the use-case of `=`'s as replacing only particular fields